src/python/grpcio_tests/tests/unit/_dns_resolver_test.py

```diff
@@ -50,7 +51,12 @@ def test_connect_loopback(self):
         # supported by custom io manager like "gevent" or "libuv".
         with grpc.insecure_channel(
                 'loopback4.unittest.grpc.io:%d' % self._port) as channel:
-            self.assertEqual(channel.unary_unary(_METHOD)(_REQUEST), _RESPONSE)
+            response = channel.unary_unary(_METHOD)(
+                _REQUEST,
+                timeout=test_constants.SHORT_TIMEOUT,
+                wait_for_ready=True,
```

Comment:
Why is this needed? Is this is needed to prevent flakiness in the test?

Reply:
The `gevent` on Windows failed in last commit, and I am not 100% sure about the root cause so I use `wait-for-ready` to de-flake.

Test log highlight:
```
Traceback (most recent call last):
  File "C:\Python35\Lib\unittest\case.py", line 59, in testPartExecutor
    yield
  File "C:\Python35\Lib\unittest\case.py", line 601, in run
    testMethod()
  File "T:\src\github\grpc\workspace_python_windows_opt_gevent\src\python\grpcio_tests\tests\unit\_dns_resolver_test.py", line 53, in test_connect_loopback
    self.assertEqual(channel.unary_unary(_METHOD)(_REQUEST), _RESPONSE)
  File "T:\src\github\grpc\workspace_python_windows_opt_gevent\py35_gevent\lib\site-packages\grpc\_channel.py", line 562, in __call__
    return _end_unary_response_blocking(state, call, False, None)
  File "T:\src\github\grpc\workspace_python_windows_opt_gevent\py35_gevent\lib\site-packages\grpc\_channel.py", line 466, in _end_unary_response_blocking
    raise _Rendezvous(state, None, None, deadline)
grpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Connect Failed"
	debug_error_string = "{"created":"@1553726691.297000000","description":"Failed to create subchannel","file":"src/core/ext/filters/client_channel/client_channel.cc","file_line":2715,"referenced_errors":[{"created":"@1553726691.297000000","description":"failed to connect to all addresses","file":"src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc","file_line":444,"referenced_errors":[{"created":"@1553726691.297000000","description":"Connect Failed","file":"src/core/ext/filters/client_channel/subchannel.cc","file_line":963,"grpc_status":14,"referenced_errors":[{"created":"@1553726691.297000000","description":"connect failed: [Errno 10061] [WinError 10061] No connection could be made because the target machine actively refused it.","file":".\src/core/lib/iomgr/gevent_util.h","file_line":33,"grpc_status":14}]}]}]}"
>
```

Reply:
One issue I think is that server is binding to `[::]` (the ipv6 loopback interface), but the client is trying to connect to the ipv4 loopback. I <i>think</i> that changing the server to bind to 127.0.0.1 would fix the issue.

I'm just worried about "wait-for-ready" masking underlying bugs....

Reply:
Most of our Python servers are binding to `[::]` and the clients are connected to `localhost`. If the platform doesn't support IPv6, I presume the `localhost` will be resolved into `127.0.0.1`, IMO this test case's setup align with other test cases.

`wait-for-ready` flag should not affect correctness, but I agree that the connection failed after the start of server is definitely weird. I want to see if the problem go away with `gevent` Windows tests.

Reply:
> `wait-for-ready` flag should not affect correctness

There are risks with using `wait-for-ready` for a test like this. If there is, for example, a bug in the resolver that causes only 1/10 DNS resolution attempts to succeed, then use of `wait-for-ready` would mask that issue and this test would always pass.... I think it's best if it can be avoided